src/Container.ts

```diff
@@ -89,8 +92,16 @@ export class Container {
             params.unshift(null);
         }
 
-        // create a new instance of the requested object
-        const objectInstance = new (type.bind.apply(type, params))();
+        let objectInstance: any;
+
+        if (factory) {
+            // Calling factory function if specified to create an instance.
+            objectInstance = factory.apply(undefined, params);
```

Comment:
If you would call `factory(params)` instead of `apply()` then all parameters would be passed as an array to the first argument of the factory function. I think it's undesired behavior? Shouldn't the elements of `params` array be mapped to the arguments of the factory function?